src/device-feature.js

```diff
@@ -0,0 +1,16 @@
+export default class {
+
+    constructor(params) {
+        this.df_name = params['df_name'];
+        this.df_type = params['df_type'];  // idf | odf
+        this.param_type = params['param_type'] ? params['param_type'] : null;
```

Comment:
這個跟 python 不同吧，我想是 `[null]`

```suggestion
        this.param_type = params['param_type'] || [null];
```